src/components/FlowListItem.vue

```diff
@@ -0,0 +1,198 @@
+          <ListItemMetaFlowRun :title="localization.info.nextRun" :flow-run="nextRun" class="flow-list-item__next-run" />
+        </div>
+      </template>
+
+      <template #action>
+        <FlowMenu size="xs" :flow="flow" show-all @delete="handleDelete" />
+      </template>
+
+      <div class="flow-list-item__content" :class="computedClasses.content" @click="toggle">
+        <p-divider class="flow-list-item__divider" />
+        <p-button
+          v-if="deploymentsCount > 0"
+          size="xs"
+          class="flow-list-item__content-toggle"
+          :class="computedClasses.toggle"
+          inset
+          icon="ChevronDownIcon"
+        />
+
+        <span v-if="deploymentsCount > 0" class="flow-list-item__content-text">
+          {{ deploymentsCount }}  {{ toPluralString(localization.info.deployment, deploymentsCount) }}
+        </span>
+
+        <span v-else class="flow-list-item__content-text-none">
+          {{ localization.info.noDeployments }}
+
+          <ExtraInfoModal :title="localization.info.noDeployments">
+            <p-markdown-renderer :text="localization.info.deploymentsEmptyStateDescription(flow.name)" />
+            <template #actions>
+              <DocumentationButton :to="localization.docs.deployments" />
+            </template>
+          </ExtraInfoModal>
+        </span>
+      </div>
+    </StateListItem>
+
+    <keep-alive v-if="canExpand">
+      <p-auto-height-transition>
+        <template v-if="expanded">
+          <slot>
+            <DeploymentList :disabled="disabled" :filter="filter" class="flow-list-item__deployments" />
+          </slot>
+        </template>
+      </p-auto-height-transition>
+    </keep-alive>
+  </div>
+</template>
+
+<script lang="ts">
+  export default {
+    name: 'FlowListItem',
+    expose: [],
+    inheritAttrs: false,
+  }
+</script>
+
+<script lang="ts" setup>
+  import { toPluralString, useAttrsStylesAndClasses } from '@prefecthq/prefect-design'
+  import { useLocalStorage, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import { DeploymentList, DocumentationButton, ExtraInfoModal, ListItemMetaFlowRun, StateListItem } from '@/components'
+  import { useNextFlowRun, useLastFlowRun, useWorkspaceApi, useWorkspaceRoutes, useComponent } from '@/compositions'
+  import { localization } from '@/localization'
+  import { DeploymentsFilter, Flow, FlowsFilter } from '@/models'
+
+  const props = defineProps<{
+    flow: Flow,
+    filter?: FlowsFilter,
+    disabled?: boolean,
+  }>()
+
+  const emit = defineEmits<{
+    (event: 'delete', value: string): void,
+  }>()
+
+  const { FlowMenu } = useComponent()
+  const { attrs, classes, styles } = useAttrsStylesAndClasses()
+  const api = useWorkspaceApi()
+  const routes = useWorkspaceRoutes()
+
+  const { value: expanded } = useLocalStorage(`flow-list-item-${props.flow.id}--expanded`, false)
```

Comment:
I think this should be session storage if anything at all. Reason being local storage and session storage have a limited amount of space. I've worked on apps in the past that saved a lot of state in local storage and ended up in a situation where some power users would get storage errors because they ran out of space. Session storage at least ensures this gets cleared out automatically when closing the app.

Reply:
Generally I try and only use local storage for things that are intended to exist forever. Basically "unless something changes in the app, this will always be used". But since this includes a user value in it (the flow id) this is likely to be removed and the local storage key won't be cleaned up.

Reply:
Makes sense to me - if we want to retain this behavior cross-session we can come up with a more robust solution at that point 👍🏻